cmd/kube-proxy/app/server.go

```diff
@@ -73,6 +74,8 @@ func (s *ProxyServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.ResourceContainer, "resource-container", s.ResourceContainer, "Absolute name of the resource-only container to create and run the Kube-proxy in (Default: /kube-proxy).")
 	fs.StringVar(&s.Kubeconfig, "kubeconfig", s.Kubeconfig, "Path to kubeconfig file with authorization information (the master location is set by the master flag).")
 	fs.Var(&s.PortRange, "proxy-port-range", "Range of host ports (beginPort-endPort, inclusive) that may be consumed in order to proxy service traffic. If unspecified (0-0) then ports will be randomly chosen.")
+	fs.Var(&s.PortRange, "proxy-port-range", "Range of host ports (beginPort-endPort, inclusive) that may be consumed in order to proxy service traffic. If unspecified (0-0) then ports will be randomly chosen.")
+	fs.IPNetVar(&s.ContainerCidr, "container-cidr", net.IPNet{}, "Cidr Range from which container IPs are allocated. If unspecified, external IPs will not be respected in the Kubernetes Service Spec.")
```

Comment:
I have spent all evening thinking about this and I am not sure what is best.  Kubelet doesn't salt this in except on the master.  Maybe we should just do like kubelet - figure out the node name and ask the master for the node's CIDR.

If you're clever you can capture it as a library, maybe?

@dchen1107 for consult